subsys/bluetooth/controller/Kconfig.ll_sw_split

```diff
@@ -583,10 +583,21 @@ config BT_CTLR_DYNAMIC_INTERRUPTS
 	  permit use of SoC's peripheral for custom use when Bluetooth is not
 	  enabled.
 
+config BT_CTLR_LTO
+	bool "Link Time Optimization"
```

Comment:
This is not a good location for this setting.

It may give the impression that LTO is only applied on this subsystem, which is actually not the case.
Enabling LTO here will enable LTO for the Zephyr build.

One may say, `of course, that should be obvious`, but in Zephyr we are actually having parts which could be using LTO while other parts would not be using it, for example for the TF-M secure build.

Reply:
I have now added `depends on ARM && BT_HCI_RAW`, this will restrict LTO to samples/application that build only the controller subsystem (aka hci_uart, hci_ipc... and so on) with a HCI transport.

And on the hci_ipc for now needs reduced code space usage in-order to support ISO features and be able to have spare space for DFU.

Reply:
Removed `&& BT_HCI_RAW`  as normal samples/application would like to use LTO.